### ISSUE TITLE
key-spacing fine-grained control issue

### ISSUE THREAD

**Issue by DarkPark** [eslint/eslint#5479]
eslint v2.3.0

Rule:

```
"key-spacing": [1, {
    "singleLine": {
        "beforeColon": false, "afterColon": true, "mode": "strict"
    },
    "multiLine": {
        "beforeColon": false, "afterColon": true, "mode": "minimum"
    }
}],
```

Code:

```js
module.exports = {
    foobar: 42,
    bat:    2 * 2
};
```

Expected result: no warnings.

Actual:
```
/home/dp/Projects/sdk/spa/plugin-css/index.js
  10:13  warning  Extra space before value for key 'bat'  key-spacing

✖ 1 problem (0 errors, 1 warning)
```

At the same time rule

```
"key-spacing": [1, {"beforeColon": false, "afterColon": true, "mode": "minimum"}],
```

gives no warnings
seems `multiLine` option is not working properly

**Comment by btmills** [comment]
Yeah, it looks like `multiLine.mode` isn't being applied correctly.

**Comment by btmills** [comment]
On [line 373](https://github.com/eslint/eslint/blob/ef341be49669230bb33c76886c666be89c8e99ee/lib/rules/key-spacing.js#L373), `isSingleLine` is being passed a `Property` node, but it should be passed the `ObjectExpression`. @DarkPark want to submit a PR?

**Comment by DarkPark** [comment]
@btmills thanks, but I have not investigated the source code so it's better to be done by someone knowledgeable.

**Comment by platinumazure** [comment]
@btmills I'll take it.
